### ISSUE TITLE
Article not displayed accurately

### ISSUE THREAD

**Issue by Inbefortus** [kiwix/kiwix-js-windows#193]
This applies to both desktop and mobile.

Kiwix JS PWA:

![Screenshot_20210916-174437_Samsung Internet](https://user-images.githubusercontent.com/71934042/133644032-60cc50be-402f-4b0a-8f3b-22a5c1d80b05.jpg)

Kiwix Android:

![Screenshot_20210916-175408_Kiwix](https://user-images.githubusercontent.com/71934042/133645062-2c80e7d1-6a71-4f6f-a8d4-326d35cc7b35.jpg)
![Screenshot_20210916-175422_Kiwix](https://user-images.githubusercontent.com/71934042/133645083-0387aa28-e5cc-478c-b870-d824954cbd72.jpg)

Original Android Wikipedia app:

![Screenshot_20210916-174546_Wikipedia](https://user-images.githubusercontent.com/71934042/133644064-aa72ca49-210b-40e5-a125-32d1e5ba138e.jpg)

**Comment by Inbefortus** [comment]
Other example:

Kiwix JS PWA:

![Screenshot_20210917-114032_Samsung Internet](https://user-images.githubusercontent.com/71934042/133762539-3443c233-f955-4b5e-a17c-549126a84e4f.jpg)

Kiwix Android:

![Screenshot_20210917-114009_Kiwix](https://user-images.githubusercontent.com/71934042/133762689-6b95e92c-905a-4853-8306-c44ebbd1a01d.jpg)

Original:

![Screenshot_20210917-114110_Samsung Internet](https://user-images.githubusercontent.com/71934042/133762760-ecfb2630-10f4-44eb-90aa-30fc41082b9f.jpg)
![20210917_114201](https://user-images.githubusercontent.com/71934042/133762797-e0d69bc1-8442-4761-8826-954677ad083b.jpg)

**Comment by Jaifroid** [comment]
Thanks for the report. I'll investigate.

**Comment by Jaifroid** [comment]
I think the second one might be caused by the transformation to desktop mode, which is a bit hacky. It searches for infoboxes to move them, but it's possible it selects too much, and includes some test it shouldn't.

**Comment by Inbefortus** [comment]
@Jaifroid If it helps, here is another example:

Kiwis JS PWA:

![20210917_155031](https://user-images.githubusercontent.com/71934042/133794279-efb3a76c-3627-473e-b774-3b69840b7957.jpg)

Original:

![20210917_155138](https://user-images.githubusercontent.com/71934042/133794438-715dce8c-e182-4356-9ac5-0b2c90658c83.jpg)

**Comment by Jaifroid** [comment]
@Inbefortus This issue is now fixed in Version 1.7.4-rc2. I've pushed this to pwa.kiwix.org, so if you have that version of the PWA, you should be able to verify (also in the dev pwa: kiwix.github.io/kiwix-js-windows/). Please ensure you are using rc2 (the app will auto-update after a restart).

The issue was an over-greedy regular expression that moves misplaced hatnotes that don't have any class or id that enables a cleaner way of moving them. So I have to use heuristics to guess common patterns. I've put in a limit to the amount the regex will scan within the `<dl>` block. The hatnotes are misplaced (in the ZIM) due to a bug that has a long-standing issue on mwOffliner: https://github.com/openzim/mwoffliner/issues/182 . Kiwix JS PWA tries to compensate for this error by scanning for hatnotes that do not appear after the title. 

Please see screenshots below. 

![2021-09-19](https://user-images.githubusercontent.com/4304337/133944522-a763d8c3-9ff6-4bed-9ac6-aa0eed9fbc99.png)

![2021-09-19 (1)](https://user-images.githubusercontent.com/4304337/133944528-b7224482-0ead-41ef-b2d6-085154db632c.png)

![2021-09-19 (2)](https://user-images.githubusercontent.com/4304337/133944529-9d3167d6-32ce-4bd1-8990-30627a45acba.png)

**Comment by Inbefortus** [comment]
@Jaifroid Nice. I can also confirm that it has been fixed!


I have a question about one of your screenshots:  In the article "Glossary of Islam" there is a small table called "Contents", what did you do to make it appear? I do not have such a table. The ZIM I use is 2021-09 English Wikipedia Endless.

**Comment by Jaifroid** [comment]
@Inbefortus The screenshot is from `wikipedia_en_all_maxi_2021-02.zim`. I've just checked the Endless version, and the table doesn't appear to be in the HTML, hence it must not have been scraped in that version. It's a pity. Below is the HTML of that table, in case @kelson42 can shed any light on why it is not being scraped post 2021-02:

```
<div role="navigation" id="toc" class="toc  hlist" aria-labelledby="tocheading" style=" text-align:left;">
<div id="toctitle" class="toctitle" style="text-align:center;display:inline-block;"><span id="tocheading" style="font-weight:bold;">Contents<span>:</span><span>&nbsp;</span></span></div>
<div style="margin:auto;white-space:nowrap;display:inline-block;">
<ul><li><a href="#top">Top</a></li>
<li><a href="#0–9">0–9</a></li>
<li><a href="#A">A</a></li>
<li><a href="#B">B</a></li>
<li><a href="#C">C</a></li>
<li><a href="#D">D</a></li>
<li><a href="#E">E</a></li>
<li><a href="#F">F</a></li>
<li><a href="#G">G</a></li>
<li><a href="#H">H</a></li>
<li><a href="#I">I</a></li>
<li><a href="#J">J</a></li>
<li><a href="#K">K</a></li>
<li><a href="#L">L</a></li>
<li><a href="#M">M</a></li>
<li><a href="#N">N</a></li>
<li><a href="#O">O</a></li>
<li><a href="#P">P</a></li>
<li><a href="#Q">Q</a></li>
<li><a href="#R">R</a></li>
<li><a href="#S">S</a></li>
<li><a href="#T">T</a></li>
<li><a href="#U">U</a></li>
<li><a href="#V">V</a></li>
<li><a href="#W">W</a></li>
<li><a href="#X">X</a></li>
<li><a href="#Y">Y</a></li>
<li><a href="#Z">Z</a> </li></ul>
<p></p>       
</div></div>
```

**Comment by Inbefortus** [comment]
@Jaifroid Ah, I see. I think that might be related to this? https://github.com/openzim/mwoffliner/issues/1514
After 2021-07, these valuable tables appear to be entirely excluded from scraping.

> I've just checked the Endless version, and the table doesn't appear to be in the HTML, hence it must not have been scraped in that version

As for that, I will create a new issue at mwoffliner.

**Comment by Jaifroid** [comment]
This fix is now in v1.7.4 of the PWA and in the latest WikiMed UWP release.
